core/src/main/kotlin/org/evomaster/core/search/gene/mongo/ObjectIdGene.kt

```diff
@@ -0,0 +1,113 @@
+package org.evomaster.core.search.gene.mongo
+
+import org.evomaster.core.logging.LoggingUtil
+import org.evomaster.core.output.OutputFormat
+import org.evomaster.core.search.gene.*
+import org.evomaster.core.search.gene.regex.*
+import org.evomaster.core.search.gene.root.CompositeFixedGene
+import org.evomaster.core.search.gene.utils.GeneUtils
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.service.mutator.genemutation.AdditionalGeneMutationInfo
+import org.evomaster.core.search.service.mutator.genemutation.SubsetGeneMutationSelectionStrategy
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * In MongoDB, documents stored in a collection require a unique _id field that acts as a primary key.
+ * ObjectIds are used as the default value for the _id field
+ */
+class ObjectIdGene(
```

Comment:
i do not understand why there is a need for such a new Gene, instead of creating it directly as Regex gene, eg, using `RegexHandler.createGeneForJVM(regex)`

Reply:
also, shouldn't there be `ObjectIdClassReplacement` for its constructor, to enable taint analysis? 
@hghianni if you are unsure what I am talking about, have a chat with @jgaleotti . or we can setup a video meeting

Reply:
btw, regarding this latter point, it could be done in a further PR / branch

Reply:
I believe the decision to create a specific gene is better. While it's true that ObjectIdGene wraps a RegexGene, I consider it more than just a regex; it models a specific type of Mongo and also has it's particular output in EJSON. About enabling taint analysis, I will talk to Juan Pablo about it.

Reply:
@hghianni  if it needs special output, then yes, could make sense to model as its own gene... but creating a new genes is not trivial, unfortunately :( we still miss a lot of documentation (+ checks and tests) to make sure important parts are not missing (eg impact analsyis). but i guess could be fine for now